src/test/lib/storage/buffer/storage_region_test.cpp

```diff
@@ -0,0 +1,115 @@
+  }
+
+  // Read from all pages and check if the data is equal to the written data
+  for (auto index = size_t{0}; index < page_ids.size(); ++index) {
+    EXPECT_NE(std::memcmp(read_pages[index].data.data(), write_pages[index].data.data(), page_ids[index].byte_count()),
+              0);
+    storage_region->read_page(page_ids[index], read_pages[index].data.data());
+    EXPECT_EQ(std::memcmp(read_pages[index].data.data(), write_pages[index].data.data(), page_ids[index].byte_count()),
+              0);
+  }
+
+  // Verifiy that the same amount of bytes was written and read
+  EXPECT_EQ(storage_region->total_bytes_written(),
+            2 * bytes_for_size_type(PageSizeType::KiB16) + bytes_for_size_type(PageSizeType::KiB32));
+  EXPECT_EQ(storage_region->total_bytes_read(),
+            2 * bytes_for_size_type(PageSizeType::KiB16) + bytes_for_size_type(PageSizeType::KiB32));
+
+  // Check that the files are deleted after the storage region is destroyed
+  storage_region = nullptr;
+  const auto files_after_cleanup = list_directory(db_path);
+  EXPECT_EQ(files_after_cleanup.size(), 0);
+}
+
+TEST_F(StorageRegionTest, WriteFailsInvalidPageID) {
+  auto page = Page{};
+  EXPECT_ANY_THROW(storage_region->write_page(INVALID_PAGE_ID, page.data.data()));
+  EXPECT_EQ(storage_region->total_bytes_written(), 0);
+}
+
+TEST_F(StorageRegionTest, ReadFailsInvalidPageID) {
+  auto page = Page{};
+  EXPECT_ANY_THROW(storage_region->read_page(INVALID_PAGE_ID, page.data.data()));
+  EXPECT_EQ(storage_region->total_bytes_read(), 0);
+}
+
+TEST_F(StorageRegionTest, WriteFailsWithUnalignedData) {
+  if constexpr (!HYRISE_DEBUG) {
+    GTEST_SKIP();
+  }
+  auto page = Page{};
+  EXPECT_ANY_THROW(storage_region->write_page(PageID{PageSizeType::KiB16, 20}, page.data.data() + 5));
+  EXPECT_EQ(storage_region->total_bytes_written(), 0);
+}
+
+TEST_F(StorageRegionTest, ReadFailsWithUnalignedData) {
+  if constexpr (!HYRISE_DEBUG) {
+    GTEST_SKIP();
+  }
+  auto page = Page{};
+  EXPECT_ANY_THROW(storage_region->read_page(PageID{PageSizeType::KiB16, 0}, page.data.data() + 5));
+  EXPECT_EQ(storage_region->total_bytes_read(), 0);
+}
+
+TEST_F(StorageRegionTest, WritingTwiceOverridesPreviousData) {
+  auto write_page = Page{};
+  auto read_page1 = Page{};
+  auto page_id = PageID{PageSizeType::KiB16, 20};
+
+  // Perform first write
+  std::memset(write_page.data.data(), 0x1, page_id.byte_count());
+  storage_region->write_page(page_id, write_page.data.data());
+  storage_region->read_page(page_id, read_page1.data.data());
+  EXPECT_EQ(std::memcmp(read_page1.data.data(), read_page1.data.data(), page_id.byte_count()), 0);
+
+  // Perform second write and check that the data was overwritten
+  auto read_page2 = Page{};
+  std::memset(write_page.data.data(), 0x2, page_id.byte_count());
+  storage_region->write_page(page_id, write_page.data.data());
+  EXPECT_NE(std::memcmp(read_page1.data.data(), read_page2.data.data(), page_id.byte_count()), 0);
```

Comment:
read_page2 is default initialized here. `read_page(page_id, read_page2)` is missing.